File: Python/scripts/Problem002.py
from functools import lru_cache
from time import time

@lru_cache
def fibonacci(n):
    if n in [0, 1]:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)



def main():
    i = answer = 0
    while (fibb_num := fibonacci(i)) < 4_000_000:
        if fibb_num % 2 == 0:
            answer += fibb_num
        i += 1

    return answer



if __name__ == '__main__':
    start = time()
    print(f"\nAnswer: { main() }")
    print(f"Time Taken: { time() - start }\n")

File: Perl/scripts/Problem002.pl
sub f_nthTerm {
    my $n = @_[0];
    my $a = 0; my $b = 1;
    for( $i = 0; $i < $n; $i++) {
        $b += $a;
        ($a, $b) = ($b, $a);
    }
    return $a;
}


$start  = time();
($i, $ans) = (1, 0);
$k = f_nthTerm($i);

while ($k < 4000000){
    $k = f_nthTerm($i);
    if ($k % 2 == 0){ $ans += $k; }
    $i++;
}

$exc_time = time() - $start;
print "\nAnswer: $ans\n";
print "Time Taken: $exc_time seconds\n\n";
